src/cloudflare/internal/ai-api.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+interface Fetcher {
+  fetch: typeof fetch
+}
+
+export type SessionOptions = {  // Deprecated, do not use this
+  extraHeaders?: object;
+};
+
+export type AiOptions = {
+  debug?: boolean;
+  prefix?: string;
+  extraHeaders?: object;
+  sessionOptions?: SessionOptions;
+};
+
+export class InferenceUpstreamError extends Error {
+  public httpCode: number;
+
+  public constructor(message: string, httpCode: number) {
+    super(message);
+    this.name = "InferenceUpstreamError";
+    this.httpCode = httpCode;
+  }
+}
+
+export class Ai {
+  private readonly fetcher: Fetcher
+
+  private options: AiOptions = {};
+  private logs: Array<string> = [];
+  public lastRequestId: string | null = null;
+
+  public constructor(fetcher: Fetcher) {
+    this.fetcher = fetcher
+  }
+
+  public async fetch(input: RequestInfo | URL, init?: RequestInit): Promise<Response> {
+    return this.fetcher.fetch(input, init)
+  }
+
+  public async run(
+    model: string,
+    inputs: Record<string, object>,
+    options: AiOptions = {}
+  ): Promise<Response | ReadableStream<Uint8Array> | object | null> {
+    this.options = options;
+    this.lastRequestId = "";
+
+    const body = JSON.stringify({
+      inputs,
+      options: {
+        debug: this.options?.debug,
+      },
+    });
+
+    const fetchOptions = {
+      method: "POST",
+      body: body,
+      headers: {
+        ...(this.options?.sessionOptions?.extraHeaders || {}),
+        ...(this.options?.extraHeaders || {}),
+        "content-encoding": "application/json",
```

Comment:
`application/json` is not a valid `content-encoding` value. Do you mean `content-type`?

Reply:
I ment `content-type`, fixed